### ISSUE TITLE
Fix Heroku builds

### ISSUE THREAD

**Issue by plusjeff** [department-of-veterans-affairs/vets-website#2872]
This is needed to support the content team's work.

Albert left last week 😢 , but even before that, the Heroku Pull Request builds weren't working. The first thing that I did was switch the builds over to va-bot from Albert's account :

![image](https://cloud.githubusercontent.com/assets/950486/18011099/9f573dd8-6b81-11e6-9ce1-b80a9a160281.png)

However, clicking the _View Deployment_ button results in this:

![image](https://cloud.githubusercontent.com/assets/950486/18011122/b8931ede-6b81-11e6-8c0a-5ac93f6e32c2.png)
 
When I look at the build log, I get the following things:

```
-----> Node.js app detected
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
       NODE_ENV=production
       NODE_MODULES_CACHE=true
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       
       Resolving node version (latest stable) via semver.io...
       Downloading and installing node 5.11.1...
       Using default npm version: 3.8.6
-----> Restoring cache
       Loading 2 from cacheDirectories (default):
       - node_modules
       - bower_components (not cached - skipping)
-----> Building dependencies
       Installing node modules (package.json + shrinkwrap)
-----> Caching build
       Clearing previous node cache
       Saving 2 cacheDirectories (default):
       - node_modules
       - bower_components (nothing to cache)
-----> Build succeeded!
       └── fsevents@1.0.14
       
-----> Discovering process types
       Procfile declares types -> web
-----> Compressing...
       Done: 142.7M
-----> Launching...
       Released v5
       https://vetsgov-pr-2866.herokuapp.com/ deployed to Heroku
```

While I don't know if this is the problem, it is noteworthy that it is using a different version of `node` and `npm` than prescribed by the README file. 

Probably the more relevant part is that I don't see where it's running the `script/heroku-serve.sh` script. Guessing this is what is causing it, but please investigate and fix.

@crwallace : I'll assign to you, but feel free to re-assign as needed. I made both you and @bshyong admins for our VA Heroku account. Let me know if you need any additional details.

**Comment by awong-dev** [comment]
Kibitz: Heroku, IIRC, has a distinction between first app build and subsequent updates. I think I screwed up the app.json or something such that it's not rebuilding or rerunning the server on updates.

Also, I would just kill scripts/heroku-server.sh and make it do `npm run watch -- -p ${PORT}` (the extra `--` is intentional).  The original script was me trying to save some cycles on the heroku server which is kinda silly.... The latter method makes heroku match the developer environment which is a better idea.

**Comment by bshyong** [comment]
Thanks @awong-dev I'll try that.

**Comment by bshyong** [comment]
I think I got things working: https://github.com/department-of-veterans-affairs/vets-website/pull/2873

Initially our build process was taking too long and too much memory. Since it is kicked off by the `web` process in the Procfile, it gets killed by Heroku.

```
2016-08-29T20:07:48.629263+00:00 heroku[web.1]: Process running mem=573M(109.7%)
2016-08-29T20:07:48.629361+00:00 heroku[web.1]: Error R14 (Memory quota exceeded)
```

I moved the build process to a Heroku `postbuild` script in `package.json`, which runs after `npm install`. 

See here for relevant changes: https://github.com/department-of-veterans-affairs/vets-website/pull/2873/commits/40ad24f55cc79721114c1e16af4bc0611748d4f4

Now the web process in Procfile is a simple http-server that runs our built app
